src/kernel/src/fs/null/vnode.rs

```diff
@@ -56,25 +85,14 @@ impl Errno for BypassError {
     }
 }
 
-fn lookup(vn: &Arc<Vnode>, td: Option<&VThread>, name: &str) -> Result<Arc<Vnode>, Box<dyn Errno>> {
-    let null_mount: &NullFs = vn.data().downcast_ref().unwrap();
-
-    let lower = null_mount.lower().unwrap().lookup(td, name)?;
-
-    let vnode = if Arc::ptr_eq(&lower, vn) {
-        vn.clone()
-    } else {
-        todo!();
-    };
-
-    Ok(vnode)
+#[derive(Debug, Error)]
+pub enum LookupError {
+    #[error("lookup failed")]
+    LookupFailed(#[source] Box<dyn Errno>),
```

Comment:
I think `LookupFromLowerFailed` are more descriptive.

Reply:
@SuchAFuriousDeath I mean the variant, not enum itself.

Reply:
oh :D